test_install/test_install.cpp

```diff
@@ -192,202 +305,177 @@ int main(int, char **)
     {
         using Bj = gko::preconditioner::Jacobi<>;
-        auto test = Bj::build().with_max_block_size(1u).on(refExec);
+        auto test = Bj::build().with_max_block_size(1u).on(exec);
     }
 
     // core/solver/bicgstab.hpp
     {
         using Solver = gko::solver::Bicgstab<>;
-        auto test = Solver::build()
-                        .with_criteria(
-                            gko::stop::Iteration::build().with_max_iters(1u).on(
-                                refExec))
-                        .on(refExec);
+        check_solver<Solver>(exec, A_raw, gko::lend(b), gko::lend(x));
     }
 
     // core/solver/cb_gmres.hpp
     {
         using Solver = gko::solver::CbGmres<>;
-        auto test = Solver::build()
-                        .with_criteria(
-                            gko::stop::Iteration::build().with_max_iters(1u).on(
-                                refExec))
-                        .on(refExec);
+        check_solver<Solver>(exec, A_raw, gko::lend(b), gko::lend(x));
     }
 
     // core/solver/cg.hpp
     {
         using Solver = gko::solver::Cg<>;
-        auto test = Solver::build()
-                        .with_criteria(
-                            gko::stop::Iteration::build().with_max_iters(1u).on(
-                                refExec))
-                        .on(refExec);
+        check_solver<Solver>(exec, A_raw, gko::lend(b), gko::lend(x));
     }
 
     // core/solver/cgs.hpp
     {
         using Solver = gko::solver::Cgs<>;
-        auto test = Solver::build()
-                        .with_criteria(
-                            gko::stop::Iteration::build().with_max_iters(1u).on(
-                                refExec))
-                        .on(refExec);
+        check_solver<Solver>(exec, A_raw, gko::lend(b), gko::lend(x));
     }
 
     // core/solver/fcg.hpp
     {
         using Solver = gko::solver::Fcg<>;
-        auto test = Solver::build()
-                        .with_criteria(
-                            gko::stop::Iteration::build().with_max_iters(1u).on(
-                                refExec))
-                        .on(refExec);
+        check_solver<Solver>(exec, A_raw, gko::lend(b), gko::lend(x));
     }
 
     // core/solver/gmres.hpp
     {
         using Solver = gko::solver::Gmres<>;
-        auto test = Solver::build()
-                        .with_criteria(
-                            gko::stop::Iteration::build().with_max_iters(1u).on(
-                                refExec))
-                        .on(refExec);
+        check_solver<Solver>(exec, A_raw, gko::lend(b), gko::lend(x));
     }
 
     // core/solver/ir.hpp
     {
         using Solver = gko::solver::Ir<>;
-        auto test = Solver::build()
-                        .with_criteria(
-                            gko::stop::Iteration::build().with_max_iters(1u).on(
-                                refExec))
-                        .on(refExec);
+        auto test =
```

Comment:
Can we also not test with IR with some inner solver ?

Reply:
We would need a special case just for IR, since the purpose isn't to cover everything but instead to make sure most stuff work, I found it not really required to test IR here. If multiple people think we should do it, I can add it.

Reply:
Yeah, I guess it is not really required as we test separately it anyway.